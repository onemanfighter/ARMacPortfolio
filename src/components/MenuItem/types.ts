export type MenuItemComponentProps = {
  text: string;
  command?: string;
  icon?: React.ReactElement;
  ariaLabel: string;
  side?: 'left' | 'right';
  onClick: () => void;
};
