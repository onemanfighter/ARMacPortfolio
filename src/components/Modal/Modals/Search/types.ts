export type SearchItem = {
  title: string;
  author: {
    firstName: string;
    lastName: string;
  };
};
