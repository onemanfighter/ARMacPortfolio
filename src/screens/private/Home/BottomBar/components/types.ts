export enum ProgramType {
  FINDER = 'finder',
  CHROME = 'chrome',
  VSCODE = 'vscode',
  TERMINAL = 'terminal',
  SPOTIFY = 'spotify',
  GITHUB = 'github',
  SETTINGS = 'settings',
}

export type IconButtonProps = {
  type: ProgramType;
};
