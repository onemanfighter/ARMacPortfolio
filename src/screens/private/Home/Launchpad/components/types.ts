import { ProgramType } from '@processStore';

export type ProgramButtonProps = {
  name: string;
  icon: string;
  isActive: boolean;
  onClickHandler: () => void;
};
