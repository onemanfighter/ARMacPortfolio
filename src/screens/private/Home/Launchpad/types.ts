import { ProgramType } from '@processStore';

export type LaunchpadProps = {};

export type ProgramData = {
  name: string;
  icon: string;
  programType: ProgramType;
};
