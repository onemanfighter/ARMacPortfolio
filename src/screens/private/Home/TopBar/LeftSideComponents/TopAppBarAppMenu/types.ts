export type MenuTreeNode = {
  title: string;
  hasDivider: boolean;
  command?: string;
  subSubtree?: MenuTreeNode[];
  action: () => void;
};

export type MenuTreeBranch = {
  title: string;
  type: 'main' | 'normal';
  items: Record<string, MenuTreeNode>;
};

export type TopAppMenuTreeStructure = {
  id: string;
  menuTree: MenuTreeBranch[];
};
