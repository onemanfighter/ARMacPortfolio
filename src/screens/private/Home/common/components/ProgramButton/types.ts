import { ProgramType } from '@processStore';

export type ProgramButtonProps = {
  name: string;
  icon: string;
  isActive: boolean;
  variant?: 'default' | 'bottomBar';
  onClickHandler: () => void;
  size?: number;
};
