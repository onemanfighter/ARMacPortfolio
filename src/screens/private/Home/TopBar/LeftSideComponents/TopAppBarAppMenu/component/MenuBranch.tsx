import { Divider, Menu, MenuButton, Text } from '@chakra-ui/react';
import { MenuItemComponent, MenuListComponent } from '@components';
import { MenuTreeBranch } from '../types';
import {
  darkModeColorSelector,
  settingsStore,
  useShallow,
} from '@settingsStore';
import _ from 'lodash';

const MenuBranch = ({ title, type, items }: MenuTreeBranch) => {
  const { iconColor } = settingsStore(useShallow(darkModeColorSelector));

  return (
    <Menu>
      <MenuButton
        fontSize={'12'}
        mx={0.2}
        px={3}
        py={0.5}
        color={iconColor}
        borderRadius={4}
        fontWeight={'600'}
        _active={{
          bg: '#ffffff4f',
        }}
        aria-label={`${_.kebabCase(title.toLowerCase())}-top-bar-button`}
      >
        <Text fontWeight={type === 'main' ? 800 : 500}>{title}</Text>
      </MenuButton>
      <MenuListComponent>
        {Object.values(items).map((item, index) => {
          if (item.subSubtree === undefined) {
            return (
              <>
                <MenuItemComponent
                  key={index + '-' + item.title}
                  text={item.title}
                  ariaLabel={
                    _.kebabCase(item.title.toLowerCase()) + '-menu-button'
                  }
                  onClick={item.action}
                  command={item.command}
                  side={'left'}
                />
                {item.hasDivider ? (
                  <Divider p={0} m={0.5} color={'GrayText'} />
                ) : null}
              </>
            );
          } else {
            return (
              <Menu key={index + '-' + item.title} placement="right-start">
                <MenuButton
                  width={'100%'}
                  borderRadius={2}
                  aria-label={`${_.kebabCase(item.title.toLowerCase())}-menu-button`}
                  _hover={{ bg: 'blue.500' }}
                  _active={{ bg: 'blue.500' }}
                >
                  <MenuItemComponent
                    text={item.title}
                    ariaLabel={item.title + '-inner-menu-item'}
                    onClick={item.action}
                    command={item.command}
                    side={'left'}
                  />
                </MenuButton>
                {item.hasDivider ? <Divider p={0} m={0.5} /> : null}
                <MenuListComponent orientation="horizontal">
                  {item.subSubtree.map((subItem, subIndex) => {
                    return (
                      <>
                        <MenuItemComponent
                          key={subIndex + '-' + subItem.title + '-sub-item'}
                          text={subItem.title}
                          ariaLabel={
                            _.kebabCase(subItem.title.toLowerCase()) +
                            '-sub-menu-item'
                          }
                          onClick={subItem.action}
                          command={subItem.command}
                          side={'left'}
                        />
                        {subItem.hasDivider ? (
                          <Divider p={0} m={0.5} color={'GrayText'} />
                        ) : null}
                      </>
                    );
                  })}
                </MenuListComponent>
              </Menu>
            );
          }
        })}
      </MenuListComponent>
    </Menu>
  );
};

export default MenuBranch;
