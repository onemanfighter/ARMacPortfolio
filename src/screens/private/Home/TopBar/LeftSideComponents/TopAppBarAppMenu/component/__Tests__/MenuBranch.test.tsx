import { fireEvent, render, screen } from '@testing-library/react';
import MenuBranch from '../MenuBranch';
import { MenuTreeNode } from '../../types';

const actionMockTest1 = jest.fn();
const actionMockTest2 = jest.fn();
const actionMockTest3 = jest.fn();
const actionMockTest4 = jest.fn();

const dummyData: Record<string, MenuTreeNode> = {
  test1: {
    title: 'test1',
    command: 'command1',
    action: actionMockTest1,
    hasDivider: false,
  },
  test2: {
    title: 'test2',
    command: 'command2',
    action: actionMockTest2,
    hasDivider: true,
  },

  test3: {
    title: 'test2',
    command: 'command2',
    action: actionMockTest3,
    hasDivider: true,
    subSubtree: [
      {
        title: 'subTree Test1',
        command: 'subTreeCommand3',
        action: actionMockTest4,
        hasDivider: true,
      },
    ],
  },
};

describe('MenuBranch', () => {
  beforeEach(() => {
    actionMockTest1.mockClear();
    actionMockTest2.mockClear();
    actionMockTest3.mockClear();
    actionMockTest4.mockClear();
  });

  it('should render with main type', () => {
    const { container } = render(
      <MenuBranch title={'Test'} type={'main'} items={dummyData} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with list open with main type', () => {
    const { container } = render(
      <MenuBranch title={'Test'} type={'main'} items={dummyData} />,
    );

    fireEvent.click(screen.getByLabelText('test-top-bar-button'));

    expect(container).toMatchSnapshot();
  });

  it('should render with normal type', () => {
    const { container } = render(
      <MenuBranch title={'Test'} type={'normal'} items={dummyData} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with list open with normal type', () => {
    const { container } = render(
      <MenuBranch title={'Test'} type={'normal'} items={dummyData} />,
    );

    fireEvent.click(screen.getByLabelText('test-top-bar-button'));

    expect(container).toMatchSnapshot();
  });

  it('should trigger item action on clicking with normal type', () => {
    const { container } = render(
      <MenuBranch title={'Test'} type={'normal'} items={dummyData} />,
    );

    fireEvent.click(screen.getByLabelText('test-1-menu-button'));

    expect(actionMockTest1).toHaveBeenCalledTimes(1);
  });

  it('should trigger item action when inner menu is clicked with normal type', () => {
    const { container } = render(
      <MenuBranch title={'Test'} type={'normal'} items={dummyData} />,
    );

    fireEvent.click(screen.getByLabelText('sub-tree-test-1-sub-menu-item'));

    expect(actionMockTest4).toHaveBeenCalledTimes(1);
  });
});
