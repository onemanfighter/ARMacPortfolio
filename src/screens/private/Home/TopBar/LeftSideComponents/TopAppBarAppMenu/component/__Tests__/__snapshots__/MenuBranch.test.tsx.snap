// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBranch should render with list open with main type 1`] = `
<div>
  <button
    aria-controls="menu-list-:r9:"
    aria-expanded="true"
    aria-haspopup="menu"
    aria-label="test-top-bar-button"
    class="chakra-menu__menu-button css-a89oud"
    data-active=""
    id="menu-button-:r9:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-ty7r4z"
      >
        Test
      </p>
    </span>
  </button>
  <div
    class="css-0"
    data-popper-escaped=""
    data-popper-placement="bottom-start"
    data-popper-reference-hidden=""
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 8px); --popper-transform-origin: top left;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:r9:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        aria-label="test1-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="0"
        id="menu-list-:r9:-menuitem-:ra:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test1
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command1
        </span>
      </button>
      <button
        aria-label="test2-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="1"
        id="menu-list-:r9:-menuitem-:rb:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test2
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command2
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-15ckxw5"
      />
      <button
        aria-controls="menu-list-:rd:"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="test2-menu-button"
        class="chakra-menu__menu-button css-p03ncx"
        id="menu-button-:rd:"
      >
        <span
          class="css-xl71ch"
        >
          <button
            aria-label="test2-inner-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="0"
            id="menu-list-:rd:-menuitem-:re:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              test2
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              command2
            </span>
          </button>
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-9li2h0"
      />
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-f6jf1x"
          id="menu-list-:rd:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            aria-label="subtreetest1-sub-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="1"
            id="menu-list-:rd:-menuitem-:rf:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              subTreeTest1
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              subTreeCommand3
            </span>
          </button>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-15ckxw5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MenuBranch should render with list open with normal type 1`] = `
<div>
  <button
    aria-controls="menu-list-:rp:"
    aria-expanded="true"
    aria-haspopup="menu"
    aria-label="test-top-bar-button"
    class="chakra-menu__menu-button css-a89oud"
    data-active=""
    id="menu-button-:rp:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-10rvbm3"
      >
        Test
      </p>
    </span>
  </button>
  <div
    class="css-0"
    data-popper-escaped=""
    data-popper-placement="bottom-start"
    data-popper-reference-hidden=""
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 8px); --popper-transform-origin: top left;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:rp:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        aria-label="test1-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="0"
        id="menu-list-:rp:-menuitem-:rq:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test1
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command1
        </span>
      </button>
      <button
        aria-label="test2-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="1"
        id="menu-list-:rp:-menuitem-:rr:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test2
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command2
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-15ckxw5"
      />
      <button
        aria-controls="menu-list-:rt:"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="test2-menu-button"
        class="chakra-menu__menu-button css-p03ncx"
        id="menu-button-:rt:"
      >
        <span
          class="css-xl71ch"
        >
          <button
            aria-label="test2-inner-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="0"
            id="menu-list-:rt:-menuitem-:ru:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              test2
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              command2
            </span>
          </button>
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-9li2h0"
      />
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-f6jf1x"
          id="menu-list-:rt:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            aria-label="subtreetest1-sub-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="1"
            id="menu-list-:rt:-menuitem-:rv:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              subTreeTest1
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              subTreeCommand3
            </span>
          </button>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-15ckxw5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MenuBranch should render with main type 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="test-top-bar-button"
    class="chakra-menu__menu-button css-a89oud"
    id="menu-button-:r1:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-ty7r4z"
      >
        Test
      </p>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        aria-label="test1-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="0"
        id="menu-list-:r1:-menuitem-:r2:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test1
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command1
        </span>
      </button>
      <button
        aria-label="test2-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="1"
        id="menu-list-:r1:-menuitem-:r3:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test2
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command2
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-15ckxw5"
      />
      <button
        aria-controls="menu-list-:r5:"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="test2-menu-button"
        class="chakra-menu__menu-button css-p03ncx"
        id="menu-button-:r5:"
      >
        <span
          class="css-xl71ch"
        >
          <button
            aria-label="test2-inner-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="0"
            id="menu-list-:r5:-menuitem-:r6:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              test2
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              command2
            </span>
          </button>
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-9li2h0"
      />
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-f6jf1x"
          id="menu-list-:r5:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            aria-label="subtreetest1-sub-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="1"
            id="menu-list-:r5:-menuitem-:r7:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              subTreeTest1
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              subTreeCommand3
            </span>
          </button>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-15ckxw5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MenuBranch should render with normal type 1`] = `
<div>
  <button
    aria-controls="menu-list-:rh:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="test-top-bar-button"
    class="chakra-menu__menu-button css-a89oud"
    id="menu-button-:rh:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-10rvbm3"
      >
        Test
      </p>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:rh:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        aria-label="test1-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="0"
        id="menu-list-:rh:-menuitem-:ri:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test1
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command1
        </span>
      </button>
      <button
        aria-label="test2-menu-button"
        class="chakra-menu__menuitem css-sfzsdj"
        data-index="1"
        id="menu-list-:rh:-menuitem-:rj:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          style="pointer-events: none; flex: 1;"
        >
          test2
        </span>
        <span
          class="chakra-menu__command css-8dknqd"
        >
          command2
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-15ckxw5"
      />
      <button
        aria-controls="menu-list-:rl:"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="test2-menu-button"
        class="chakra-menu__menu-button css-p03ncx"
        id="menu-button-:rl:"
      >
        <span
          class="css-xl71ch"
        >
          <button
            aria-label="test2-inner-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="0"
            id="menu-list-:rl:-menuitem-:rm:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              test2
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              command2
            </span>
          </button>
        </span>
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-9li2h0"
      />
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-f6jf1x"
          id="menu-list-:rl:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            aria-label="subtreetest1-sub-menu-item"
            class="chakra-menu__menuitem css-sfzsdj"
            data-index="1"
            id="menu-list-:rl:-menuitem-:rn:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              style="pointer-events: none; flex: 1;"
            >
              subTreeTest1
            </span>
            <span
              class="chakra-menu__command css-8dknqd"
            >
              subTreeCommand3
            </span>
          </button>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-15ckxw5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
